File: ios/Classes/TencentMapPlugin.swift
import Flutter
import UIKit

public class TencentMapPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    registrar.register(TencentMapFactory(registrar: registrar), withId: "tencent_map_flutter")
    _TencentMapSdkApi.setup(registrar: registrar)
  }
}

File: android/src/main/kotlin/com/morbit/tencent_map_flutter/TencentMapPlugin.kt
package com.morbit.tencent_map_flutter

import androidx.annotation.NonNull
import io.flutter.embedding.engine.plugins.FlutterPlugin

class TencentMapPlugin : FlutterPlugin {
  override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    binding.platformViewRegistry.registerViewFactory("tencent_map_flutter", TencentMapFactory(binding))
    TencentMapSdkApi.setup(binding)
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}
